cln_lspd.go

```diff
@@ -0,0 +1,281 @@
+	log.Printf("htlc_accepted called\n")
+	onion := event.Onion
+
+	log.Printf("htlc: %v\nchanID: %v\nincoming amount: %v\noutgoing amount: %v\nincoming expiry: %v\noutgoing expiry: %v\npaymentHash: %v\nonionBlob: %v\n\n",
+		event.Htlc,
+		onion.ShortChannelId,
+		event.Htlc.AmountMilliSatoshi, //with fees
+		onion.ForwardAmount,
+		event.Htlc.CltvExpiryRelative,
+		event.Htlc.CltvExpiry,
+		event.Htlc.PaymentHash,
+		onion,
+	)
+
+	paymentHash, paymentSecret, destination, incomingAmountMsat, outgoingAmountMsat, fundingTxID, fundingTxOutnum, err := paymentInfo([]byte(event.Htlc.PaymentHash))
+	if err != nil {
+		log.Printf("paymentInfo(%v)\nfundingTxOutnum: %v\n error: %v", event.Htlc.PaymentHash, fundingTxOutnum, err)
+	}
+	var zeroConfFundingTxID string
+	//var zeroConfFundingTxOutnum uint32
+	if hex.EncodeToString(paymentSecret) != "" {
+		if hex.EncodeToString(fundingTxID) == "" {
+			if hex.EncodeToString(paymentHash) == event.Htlc.PaymentHash {
+
+				zeroConfFundingTxID, _, err = clnOpenChannel(clientcln, hex.EncodeToString(paymentHash), hex.EncodeToString(destination), incomingAmountMsat)
+				log.Printf("openclnOpenChannelChannel(%v, %v) err: %v", destination, incomingAmountMsat, err)
+				if err != nil {
+					log.Printf(" clnOpenChannel error: %v", err)
+				}
+
+			} else { //probing
+				failureCode := "TEMPORARY_CHANNEL_FAILURE"
+				if err := clnIsConnected(clientcln, hex.EncodeToString(destination)); err == nil {
+					failureCode = "INCORRECT_OR_UNKNOWN_PAYMENT_DETAILS"
+				}
+				log.Printf(" failureCode %v , Error %v", failureCode, err)
+			}
+
+			return clnResumeOrCancel(clientcln, hex.EncodeToString(destination), zeroConfFundingTxID, hex.EncodeToString(paymentHash), uint64(outgoingAmountMsat), 99, event)
+
+		}
+	}
+
+	return event.Continue(), nil
+}
+
+func clnGetChannel(clientcln *glightning.Lightning, destination string, fundingTxID string) string {
+	obj, err := clientcln.ListFunds()
+	if err != nil {
+		log.Printf("clientcln.ListChannels(%v) error: %v", destination, err)
+		return ""
+	}
+	for _, c := range obj.Channels {
+		log.Printf("getChannel(%v): %v", destination, c.ShortChannelId)
+		if c.FundingTxId == fundingTxID && c.Connected {
+			chanid := c.ShortChannelId
+			if err != nil {
+				log.Printf("strconv.ParseUint(c.ShortChannelId, 10, 64) error: %v", err)
+			}
+			return chanid
+		}
+	}
+	log.Printf("No channel found: getChannel(%v)", destination)
+	return ""
+}
+func clnResumeOrCancel(clientcln *glightning.Lightning, destination string, fundingTxID string, paymentHash string, outgoingAmountMsat uint64, riskfactor float32, event *glightning.HtlcAcceptedEvent) (*glightning.HtlcAcceptedResponse, error) {
+	//deadline := time.Now().Add(10 * time.Second)
+
+	for {
+		chanID := clnGetChannel(clientcln, destination, fundingTxID)
+
+		if chanID != "" {
+			log.Printf("channel opended successfully chanID: %v)", chanID)
+			/*
+				err := insertChannel(chanID, channelPoint, destination, time.Now())
```

Comment:
why is this commented? don't we need to add the channel here?

Reply:
chanID here is a short channel id,  is a string and what we are storing in the database is uint64. this is something I need to confirm.